client/state/data-layer/wpcom/domains/nameservers/index.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { http } from 'calypso/state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'calypso/state/data-layer/wpcom-http/utils';
+import { DOMAIN_NAMESERVERS_FETCH, DOMAIN_NAMESERVERS_UPDATE } from 'calypso/state/action-types';
+import { errorNotice, successNotice } from 'calypso/state/notices/actions';
+import {
+	fetchNameserversSuccess,
+	fetchNameserversFailure,
+	updateNameserversSuccess,
+} from 'calypso/state/domains/nameservers/actions';
+import { registerHandlers } from 'calypso/state/data-layer/handler-registry';
+
+export const fetchDomainNameservers = ( action ) =>
+	http(
+		{
+			apiVersion: '1.1',
+			method: 'GET',
+			path: `/domains/${ action.domainName }/nameservers/`,
+		},
+		action
+	);
+
+export const fetchDomainNameserversSuccess = ( { domainName }, nameservers ) =>
+	fetchNameserversSuccess( domainName, nameservers );
+
+export const fetchDomainNameserversError = ( { domainName } ) =>
+	fetchNameserversFailure( domainName );
+
+export const updateDomainNameservers = ( action ) =>
+	http(
+		{
+			apiVersion: '1.1',
+			method: 'POST',
+			path: `/domains/${ action.domainName }/nameservers/`,
+			body: {
+				nameservers:
+					action.nameservers?.map( ( nameserver ) => ( {
```

Comment:
Can `action.nameservers` even be `null`? Maybe in case when they haven't been loaded from server yet in the React UI component, but then the update UI should be disabled, right?

Reply:
This is a remnant from an older version of the code. I think we'll always have `nameservers`, so I've removed the optional chaining in f4ca7bb2c4d7fb6413885664f9875457a539f13d